### PR TITLE
adding exception details to FileLogger

### LINKED ISSUE #3481: LanguageWorker Not logging syntax errors
Syntax errors coming from language workers (e.g. Node/Python) aren't being logged to our file logs (function and host). Since the file logs drive portal log streaming, this means that users won't get syntax errors during portal development.

The error does make its way back from the worker through the worker channel and you can see the error in the InvocationResponse in the debugger [here](https://github.com/Azure/azure-functions-host/blob/dev/src/WebJobs.Script/Rpc/LanguageWorkerChannel.cs#L435) but we don't log this error.

### PR DESCRIPTION
**Pull request by brettsam**
Fixes #3481

Replacement for #3482

The FileLogger wasn't logging any exception details. This is a port of the v1 behavior from: https://github.com/Azure/azure-functions-host/blob/v1.x/src/WebJobs.Script/Diagnostics/FileTraceWriter.cs#L203-L222

### CODE REVIEW

File: src/WebJobs.Script/Diagnostics/FileLogger.cs [hunk 1]
```diff
@@ -62,6 +63,28 @@ public void Log<TState>(LogLevel logLevel, EventId eventId, TState state, Except
...
+                    // we want to minimize the stack traces for function invocation
+                    // failures, so we drill into the very inner exception, which will
+                    // be the script error
+                    Exception actualException = exception;
+                    while (actualException.InnerException != null)
+                    {
+                        actualException = actualException.InnerException;
+                    }
+
+                    formattedMessage += $"{Environment.NewLine}{actualException.Message}";
```
Comment (pragnagopa) [inline src/WebJobs.Script/Diagnostics/FileLogger.cs]: Full stack trace is useful. Especially for out-of-proc languages where there are compilation errors. Thoughts?

Comment (brettsam) [reply src/WebJobs.Script/Diagnostics/FileLogger.cs]: They won't want the .NET stack, though. That'll just show the executor, invoker, etc. I suspect that whatever message an out-of-proc language logs here will include the stack for that message. I'll give it a test with node really quick.

Comment (pragnagopa) [reply src/WebJobs.Script/Diagnostics/FileLogger.cs]: You are right. I just verified with your changes
```
Result: Failure
Exception: ReferenceError: coext is not defined
Stack: ReferenceError: coext is not defined
    at module.exports (D:\pgopaGit\azure-functions-host\sample\Node\HttpTrigger\index.js:4:5)
    at WorkerChannel.invocationRequest (D:\pgopaGit\azure-functions-host\src\WebJobs.Script.WebHost\bin\Debug\netcoreapp2.1\workers\node\worker-bundle.js:28843:26)
    at ClientDuplexStream.WorkerChannel.eventStream.on (D:\pgopaGit\azure-functions-host\src\WebJobs.Script.WebHost\bin\Debug\netcoreapp2.1\workers\node\worker-bundle.js:28733:30)
    at ClientDuplexStream.emit (events.js:182:13)
    at addChunk (_stream_readable.js:283:12)
    at readableAddChunk (_stream_readable.js:264:11)
    at ClientDuplexStream.Readable.push (_stream_readable.js:219:10)
    at Object.onReceiveMessage (D:\pgopaGit\azure-functions-host\src\WebJobs.Script.WebHost\bin\Debug\netcoreapp2.1\workers\node\worker-bundle.js:43673:19)
    at InterceptingListener.module.exports.InterceptingListener.recvMessageWithContext (D:\pgopaGit\azure-functions-host\src\WebJobs.Script.WebHost\bin\Debug\netcoreapp2.1\workers\node\worker-bundle.js:43000:19)
    at D:\folder\azure-functions-host\src\WebJobs.Script.WebHost\bin\Debug\netcoreapp2.1\workers\node\worker-bundle.js:43099:14
```

Comment (brettsam) [reply src/WebJobs.Script/Diagnostics/FileLogger.cs]: I just wrote some nonsense in a js script and ran it. Before this change it looks like this in the file log:
```
2018-09-19T17:02:38.394 [Error] Executed 'Functions.HttpTrigger' (Failed, Id=74640518-a61d-4806-abcb-00676be13dca)
```

Now it looks like:
```
2018-09-19T16:59:52.268 [Error] Executed 'Functions.HttpTrigger' (Failed, Id=15486ec0-7fcd-4d41-8abb-ed75158ec90b)
Result: Failure
Exception: ReferenceError: gds is not defined
Stack: ReferenceError: gds is not defined
    at module.exports (C:\git\brettsam\azure-functions-host\sample\Node\HttpTrigger\index.js:10:5)
    at WorkerChannel.invocationRequest (c:\git\brettsam\azure-functions-host\src\WebJobs.Script.WebHost\bin\Debug\netcoreapp2.1\workers\node\worker-bundle.js:28843:26)
    at ClientDuplexStream.WorkerChannel.eventStream.on (c:\git\brettsam\azure-functions-host\src\WebJobs.Script.WebHost\bin\Debug\netcoreapp2.1\workers\node\worker-bundle.js:28733:30)
    at ClientDuplexStream.emit (events.js:182:13)
    at addChunk (_stream_readable.js:283:12)
    at readableAddChunk (_stream_readable.js:264:11)
    at ClientDuplexStream.Readable.push (_stream_readable.js:219:10)
    at Object.onReceiveMessage (c:\git\brettsam\azure-functions-host\src\WebJobs.Script.WebHost\bin\Debug\netcoreapp2.1\workers\node\worker-bundle.js:43673:19)
    at InterceptingListener.module.exports.InterceptingListener.recvMessageWithContext (c:\git\brettsam\azure-functions-host\src\WebJobs.Script.WebHost\bin\Debug\netcoreapp2.1\workers\node\worker-bundle.js:43000:19)
    at c:\git\brettsam\azure-functions-host\src\WebJobs.Script.WebHost\bin\Debug\netcoreapp2.1\workers\node\worker-bundle.js:43099:14
```

**Pull request merged**